base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -230,6 +230,14 @@ where B: BlockchainBackend
             txn.set_horizon_data(kernel_sum, utxo_sum);
             blockchain_db.write(txn)?;
             blockchain_db.store_pruning_horizon(config.pruning_horizon)?;
+        } else if blockchain_db.get_chain_metadata()?.pruning_horizon() == 0 {
+            // Check the genesis block in the DB.
+            if !blockchain_db.block_exists(genesis_block.accumulated_data().hash.clone())? {
+                error!(target: LOG_TARGET, "Stored genesis block mismatch with the binary");
```

Comment:
Please add the expected hash of the genesis block, and the hash of the block in the db to this log